src/plugins/janus_streaming.c

```diff
@@ -4423,6 +4444,30 @@ static json_t *janus_streaming_process_synchronous_request(janus_streaming_sessi
 			json_object_set_new(info, "id", string_ids ? json_string(mp->id_str) : json_integer(mp->id));
 			gateway->notify_event(&janus_streaming_plugin, session ? session->handle : NULL, info);
 		}
+
+		/* Also notify viewers */
+		if (send_edited_event) {
+			json_t *info = json_object();
+			json_object_set_new(info, "event", json_string("edited"));
+			json_object_set_new(info, "id", string_ids ? json_string(mp->id_str) : json_integer(mp->id));
+			if(mp->metadata)
+				json_object_set_new(info, "metadata", json_string(mp->metadata));
+			GList *viewer = g_list_first(mp->viewers);
+			while(viewer) {
+				janus_streaming_session *session = (janus_streaming_session *)viewer->data;
```

Comment:
I think you're shadowing the existing `session` variable in `janus_streaming_process_synchronous_request` by calling it the same way. Better to use a different name, e.g., `janus_streaming_session *s`, which is what we do in the management of other synchronous requests that result in notifications.

Reply:
Variable renamed.